dockerfiles/image.sh

```diff
@@ -0,0 +1,306 @@
+}
+
+run() {
+    local cmd=$1
+    docker run --rm \
+        --mount source=$ROOT_DIR,target=/hpcc-dev/HPCC-Platform-local,type=bind,readonly \
+        --mount source=hpcc_src,target=/hpcc-dev/HPCC-Platform,type=volume \
+        --mount source="$ROOT_DIR/.git",target=/hpcc-dev/HPCC-Platform/.git,type=bind \
+        --mount source=$HPCC_BUILD,target=/hpcc-dev/build,type=volume \
+        build-$BUILD_OS:$GITHUB_REF \
+            "cd /hpcc-dev/HPCC-Platform && \
+            $cmd"
+}
+
+init_hpcc_src() {
+    echo "  --- Init hpcc_src volume ---"
+    if ! docker volume ls -q -f name=hpcc_src | grep -q hpcc_src; then
+        echo "  --- git reset ---"
+        run "git reset --hard --recurse-submodules"
+    fi
+}
+
+reconfigure() {
+    echo "  --- Clean cmake cache ---"
+    init_hpcc_src
+    run "rm -rf /hpcc-dev/HPCC-Platform/vcpkg/vcpkg && \
+        rm -rf /hpcc-dev/build/CMakeCache.txt CMakeFiles"
+}
+
+configure() {
+    local options=$1
+    echo "  --- cmake config $options ---"
+    run "cmake -S /hpcc-dev/HPCC-Platform -B /hpcc-dev/build $options"
+}
+
+fetch_build_type() {
+    echo $(run "grep 'CMAKE_BUILD_TYPE:' /hpcc-dev/build/CMakeCache.txt | cut -d '=' -f 2")
+}
+
+calc_diffs() {
+    init_hpcc_src
+    echo "  --- Calc diff ---"
+    run "git ls-files --modified --exclude-standard" > rsync_include.txt
+    run "rsync -av --delete --delete-after --files-from=/hpcc-dev/HPCC-Platform-local/rsync_include.txt /hpcc-dev/HPCC-Platform-local/ /hpcc-dev/HPCC-Platform/"
+
+    pushd $ROOT_DIR >/dev/null
+    git status --short | grep '^ M' | cut -c4- > rsync_include.txt
+    echo "$GITHUB_REF" > tmp.txt
+    while read file; do
+        if [ -f "$file" ]; then
+            md5sum "$file" | cut -d ' ' -f 1 >> tmp.txt
+        fi
+    done < rsync_include.txt
+    local crc=$(md5sum tmp.txt | cut -d ' ' -f 1)
+    rm tmp.txt
+    popd >/dev/null
+    echo $crc
+}
+
+sync_files() {
+    echo "  --- Sync files ---"
+    run "rsync -av --delete --files-from=/hpcc-dev/HPCC-Platform-local/rsync_include.txt /hpcc-dev/HPCC-Platform-local/ /hpcc-dev/HPCC-Platform/ && 
+        git submodule update --init --recursive"
+}
+
+check_cache() {
+    local label=$1
+    local crc=$2
+    echo "  --- Check cache hpccsystems/$label:$GITHUB_BRANCH-$crc ---"
+    image_count=$(docker images --quiet hpccsystems/$label:$GITHUB_BRANCH-$crc | wc -l)
+    if [ $image_count -gt 0 ]; then
+        echo "--- Image already exists  --- "
+        echo "docker run --entrypoint /bin/bash -it hpccsystems/$label:$GITHUB_BRANCH-$crc"
+        echo "hpccsystems/$label:$GITHUB_BRANCH-$crc"
+        exit 0
+    fi
+}
+
+build() {
+    if [ "$MODE" = "release" ]; then
+        local label=platform-core
+        local base=ubuntu:jammy-20230308
+        local build_type="RelWithDebInfo"
+        local cmake_options="-DCMAKE_BUILD_TYPE=$build_type -DCPACK_STRIP_FILES=ON"
+    elif [ "$MODE" = "debug" ]; then
+        local label=platform-core-debug
```

Comment:
minor: it's going to be slightly annoying that the debug version has a different image name.
In the past the tag would be different, but incr.sh would always build "platform-core".

Being different means startall.sh will not automatically pick up the latest "platform-core-debug" (it's only looking for platform-core).